api/src/main/java/javax/enterprise/inject/spi/builder/BeanAttributesConfigurator.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package javax.enterprise.inject.spi.builder;
+
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.BeanAttributes;
+import javax.enterprise.inject.spi.ProcessBeanAttributes;
+import javax.enterprise.util.TypeLiteral;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+/**
+ * This API is an helper to configure a new {@link BeanAttributes} instance.
+ * CDI container must provides an implementation of this interface.
+ *
+ * BeanAttributesConfigurator is not reusable.
+ *
+ * This configurator is not thread safe and shall not be used concurrently.
+ *
+ * @see ProcessBeanAttributes#configureBeanAttributes()
+ * @param <T> the class of the bean instance
+ * @author Antoine Sabot-Durand
+ * @since 2.0
+ */
+public interface BeanAttributesConfigurator<T> {
+    /**
+     * Read the information from the given annotated type. All relevant information is overwritten.
+     *
+     * @param type class to read information from
+     * @return self
+     */
+    <U extends T> BeanAttributesConfigurator<U> read(AnnotatedType<U> type);
```

Comment:
We should probably also add `AnnotatedMember<U> member` variant for producer methods and fields?